libs/common/include/launchdarkly/bindings/c/config/builder.h

```diff
@@ -1,597 +1,16 @@
- */
-LD_EXPORT(void) LDLoggingCustomBuilder_Free(LDLoggingCustomBuilder b);
-
-/**
- * Sets a custom log backend.
- * @param b Custom logging builder. Must not be NULL.
- * @param backend The backend to use for logging. Ensure the backend was
- * initialized with LDLogBackend_Init.
- */
-LD_EXPORT(void)
-LDLoggingCustomBuilder_Backend(LDLoggingCustomBuilder b,
-                               struct LDLogBackend backend);
-
-/**
- * Configures the SDK with custom logging.
- * @param b  Client config builder. Must not be NULL.
- * @param custom_builder The custom logging builder. Must not be NULL.
- */
-LD_EXPORT(void)
-LDClientConfigBuilder_Logging_Custom(LDClientConfigBuilder b,
-                                     LDLoggingCustomBuilder custom_builder);
-
-/**
- * Creates a new builder for a custom, user-provided persistence.
- *
- * If not passed into the config builder, must be manually freed with
- * LDPersistenceCustomBuilder_Free.
- * @return New builder.
- */
-LD_EXPORT(LDPersistenceCustomBuilder) LDPersistenceCustomBuilder_New();
-
-/**
- * Frees a custom persistence builder. Do not call if the builder was consumed
- * by the config builder.
- * @param b Builder to free.
- */
-LD_EXPORT(void) LDPersistenceCustomBuilder_Free(LDPersistenceCustomBuilder b);
-
-/**
- * Sets a custom persistence implementation.
- * @param b Custom persistence builder. Must not be NULL.
- * @param impl The implementation to use for persistence. Ensure the
- * implementation was initialized with LDPersistence_Init.
- */
-LD_EXPORT(void)
-LDPersistenceCustomBuilder_Implementation(LDPersistenceCustomBuilder b,
-                                          struct LDPersistence impl);
-
-/**
- * Configures the SDK with custom persistence.
- * @param b  Client config builder. Must not be NULL.
- * @param custom_builder The custom persistence builder. Must not be NULL.
- * @return
- */
-LD_EXPORT(void)
-LDClientConfigBuilder_Persistence_Custom(
-    LDClientConfigBuilder b,
-    LDPersistenceCustomBuilder custom_builder);
-
-/**
- * Disables persistence.
- * @param b  Client config builder. Must not be NULL.
- */
-LD_EXPORT(void)
-LDClientConfigBuilder_Persistence_None(LDClientConfigBuilder b);
-
-/**
- * Creates an LDClientConfig. The LDClientConfigBuilder is consumed.
- * On success, the config will be stored in out_config; otherwise,
- * out_config will be set to NULL and the returned LDStatus will indicate
- * the error.
- * @param builder Builder to consume. Must not be NULL.
- * @param out_config Pointer to where the built config will be
- * stored. Must not be NULL.
- * @return Error status on failure.
- */
-LD_EXPORT(LDStatus)
-LDClientConfigBuilder_Build(LDClientConfigBuilder builder,
-                            LDClientConfig* out_config);
-
-/**
- * Frees the builder; only necessary if not calling Build.
- * @param builder Builder to free.
- */
-LD_EXPORT(void)
-LDClientConfigBuilder_Free(LDClientConfigBuilder builder);
-
-#ifdef __cplusplus
-}
-#endif
```

Comment:
So, this file still exists but just includes the new one with a compiler warning. I'm not sure we actually need to do this to remain backwards compatible in the traditional sense, but it is somewhat nicer than a build just breaking.

Why is this an issue at all? Because the main `sdk.h` binding file doesn't include the config builder bindings, since none of the types in that file need it. Users need to include it themselves. Could revisit that decision.